src/pe_source/dnstwistscript.py

```diff
@@ -0,0 +1,253 @@
+
+    # Check IPv6
+    if "dns_aaaa" not in dom:
+        dom["dns_aaaa"] = [""]
+    elif str(dom["dns_aaaa"][0]) == "!ServFail":
+        dom["dns_aaaa"] = [""]
+    else:
+        domain_aaaa = str(dom["dns_aaaa"][0])
+        # Check IP in Blocklist API
+        response = requests.get(
+            "http://api.blocklist.de/api.php?ip=" + domain_aaaa
+        ).content
+        if str(response) != "b'attacks: 0<br />reports: 0<br />'":
+            malicious = True
+            attacks = int(str(response).split("attacks: ")[1].split("<")[0])
+            reports = int(str(response).split("reports: ")[1].split("<")[0])
+        try:
+            results = json.loads(dshield.ip(damain_aaaa"), return_format=dshield.JSON))
+            threats = results["ip"]["threatfeeds"]
+            dshield_attacks = int(results["ip"].get("attacks", 0))
+            malicious = True
+            dshield_count = len(threats)
+        except KeyError:
+            dshield_attacks = 0
+            dshield_count = 0
+
+    # Clean-up other fields
+    if "ssdeep_score" not in dom:
+        dom["ssdeep_score"] = ""
+    if "dns_mx" not in dom:
+        dom["dns_mx"] = [""]
+    if "dns_ns" not in dom:
+        dom["dns_ns"] = [""]
+
+    # Ignore duplicates
+    permutation = dom["domain"]
+    if permutation in perm_list:
+        return None, perm_list
+    else:
+        perm_list.append(permutation)
+
+    domain_dict = {
+        "blocklist_attack_count": attacks,
+        "blocklist_report_count": reports,
+        "data_source_uid": source_uid,
+        "date_active": date,
+        "domain_permutation": dom["domain"],
+        "dshield_attack_count": dshield_attacks,
+        "dshield_record_count": dshield_count,
+        "fuzzer": dom["fuzzer"],
+        "ipv4": domain_a,
+        "ipv6": domain_aaaa,
+        "mail_server": dom["dns_mx"][0],
+        "malicious": malicious,
+        "name_server": dom["dns_ns"][0],
+        "organizations_uid": pe_org_uid,
+        "ssdeep_score": dom["ssdeep_score"],
+        "sub_domain_uid": sub_domain_uid,
+    }
+    return domain_dict, perm_list
+
+
+def execute_dnstwist(root_domain, test=0):
+    """Run dnstwist on each root domain."""
+    pathtoDict = f"{pathlib.Path(__file__).parent.resolve()}/data/common_tlds.dict"
+    dnstwist_result = dnstwist.run(
+        domain=root_domain,
+        format="json",
+        registered=True,
+        threads=8,
+        tld=pathtoDict,
+    )
+    if test == 1:
+        return dnstwist_result
+    finalorglist = dnstwist_result + []
+    for dom in dnstwist_result:
+        if ("tld-swap" not in dom["fuzzer"]) and ("original" not in dom["fuzzer"]):
+            LOGGER.info("Running again on %s", dom["domain"])
+            finalorglist += dnstwist.run(
+                domain=dom["domain"],
+                format="json",
+                registered=True,
+                threads=8,
+                tld=pathtoDict,
+            )
+    return finalorglist
```

Comment:
I think this should accomplish what you want to do here without needing a separate `finalorglist` variable, though you should test it to confirm:
```suggestion
    for dom in dnstwist_result.copy():
        if ("tld-swap" not in dom["fuzzer"]) and ("original" not in dom["fuzzer"]):
            LOGGER.info("Running again on %s", dom["domain"])
            dnstwist_result += dnstwist.run(
                domain=dom["domain"],
                format="json",
                registered=True,
                threads=8,
                tld=pathtoDict,
            )
    return dnstwist_result
```

Reply:
Fixed.

Reply:
Latest Commit uses dnstwist_result.copy() instead of creating a new variable

Reply:
commit acf2a7a446b4ae09e20473aee8ddfbbcc3afe83a